### ISSUE TITLE
Titanic Project 401 (UE5) crashing

### ISSUE THREAD

**Issue by Hadrianneue** [HansKristian-Work/vkd3d-proton#1451]
The post office is the only level that loads, every other level crashes with the following error message:
![image](https://user-images.githubusercontent.com/24306541/223004738-1ca9334e-1348-4b39-be1c-0f8dea1e6de0.png)

### Software information
Titanic Project 401 [page](https://titanichg.com/project-401)

### System information
- GPU: RX 6500 XT 4GB
- Driver: Mesa 23.1.0-devel (git-61663859bc)
- Wine version: Wine-GE 7-37
- VKD3D-Proton version: 2.8

### Log files
```
0170:fixme:rs_desc_from_d3d12: Ignoring MultisampleEnable 0x1.
01dc:info:dxgi_vk_swap_chain_recreate_swapchain_in_present_task: Got 3 swapchain images.
01f8:fixme:d3d12_device_SetResidencyPriority: iface 0000000027740030, object_count 1, objects 000000003da1ebb8, priorities 000000003da1ebc0 stub!
01f4:fixme:d3d12_device_Evict: iface 0000000027740030, object_count 32, objects 000000005010e200 stub!
01f4:fixme:d3d12_device_EnqueueMakeResident: iface 0000000027740030, flags 0, num_objects 2, objects 000000008de5d3c0, fence_to_signal 0000000000d70ad0, fence_value_to_signal 1 stub!
0178:fixme:vk_samples_from_dxgi_sample_desc: Unhandled sample count 0.
0174:fixme:vk_samples_from_dxgi_sample_desc: Unhandled sample count 0.
Monitored process exited.
Initial process has exited (return code: 768)
Exit with return code 768

```

**Comment by adolfintel** [comment]
Same problem here with the current master

**Comment by doitsujin** [comment]
Can you please post a full Proton log? The snippet doesn't point to there being any issue on our end and "shader compilation" can either be d3dcompiler memes or some actual d3d12 problem.

**Comment by Etaash-mathamsetty** [comment]
could someone try using native d3dcompiler

**Comment by adolfintel** [comment]
@Hadrianneue I can try it. Do you know which version of d3dcompiler it needs?

**Comment by Hadrianneue** [comment]
seems to be working fine now, no errors, i have no idea why....

**Comment by lucilanga** [comment]
> seems to be working fine now, no errors, i have no idea why....

how did you run it ?

**Comment by Hadrianneue** [comment]
> > seems to be working fine now, no errors, i have no idea why....
> 
> how did you run it ?

latest vkd3d-git and mesa git, but i have no idea for how long this has been working, i just decided to re-test it now

**Comment by adolfintel** [comment]
I can confirm that this works now with the latest vkd3d and mesa-git. I'm not really an expert but if I had to guess, I'd say it's that commit that implemented ROV in mesa that fixed this and other games a few weeks ago.

**Comment by Hadrianneue** [comment]
> @Hadrianneue This MR https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/22250

i did a build by commenting out references to that extension on radv_physical_device.c, vulkaninfo shows its false, i'm not a 100% its disabled but game still runs, tried amdvlk and it runs on latest vkd3d, try even the older vkd3d 2.8 (git release) and game still runs...

maybe its something on RADV but idk

**Comment by adolfintel** [comment]
Do a git bisect on mesa: find a commit that doesn't work and start bisecting from there. If you can't find one prior to when the issue was reported, then it's vkd3d that fixed this issue and you should do the bisect on this it instead.

**Comment by Hadrianneue** [comment]
> Do a git bisect on mesa: find a commit that doesn't work and start bisecting from there. If you can't find one prior to when the issue was reported, then it's vkd3d that fixed this issue and you should do the bisect on this project instead.

yeah im aware, but my setup with aur makes doing that a PITA...

**Comment by Hadrianneue** [comment]
> Well, it's not ideal, but if you want to do it through your aur helper, you can just modify the pkgbuild file to build a specific commit. It's kinda stupid and very manual but if you don't want to build mesa manually it's better than nothing.
> 
> vkd3d on the other hand is super easy to build and bisect.

i'll try it then later on today (tomorrow) after work, it shouldn't take too long.

**Comment by fakhraldin** [comment]
Hi, any news about this? I got the exact same error message after crashing. Only the level post office does work. 
My system setup is Fedora 38 with following specs.:
    GPU: RX 6000 8GB
    Driver: Mesa 23.1.3
    Wine version: lutris-GE-Proton8-10
    VKD3D-Proton version: 2.9
    [winelog-TitanicDemo401V2.txt]
(https://github.com/HansKristian-Work/vkd3d-proton/files/12013975/winelog-TitanicDemo401V2.txt)

I get the same error message on another UE5 (v5.1) based Demo here.:
https://github.com/HansKristian-Work/vkd3d-proton/issues/1599

and another issue based on UEv5.2 here.:
https://github.com/HansKristian-Work/vkd3d-proton/issues/1598

**Comment by Hadrianneue** [comment]
found it, fixed on mesa: https://gitlab.freedesktop.org/mesa/mesa/-/commit/91505132423bcf8949034ef277b8bee4e1936855

@fakhraldin update to mesa-git if possible, it will likely fix your issue with UE 5.2 demo too.

**Comment by fakhraldin** [comment]
@Hadrianneue I installed latest mesa-git and the UE 5.1 Demo works now, as well as two other UE5.(1) based games. 
One of them started only with parameter -vulkan before. Now both run with -dx12. That hints to a vkd3d fix.

The UE5.2 Demo though starts sometimes now, but crashes after a couple of seconds. 
Thank you for your testing and feedback.

PS: maybe also related to mesa fixes or maybe due to vkd3d fixes, the UE4 game "stray" with parameter -dx12 works now too.
